lib/openfoodfacts.dart

```diff
@@ -307,7 +307,20 @@ class OpenFoodAPIClient {
   }) async {
     final Response response = await configuration.getResponse(user, queryType);
     final String jsonStr = _replaceQuotes(response.body);
-    final SearchResult result = SearchResult.fromJson(json.decode(jsonStr));
+    final SearchResult result;
+    try {
+      result = SearchResult.fromJson(json.decode(jsonStr));
+    } on FormatException {
+      if (jsonStr.allMatches('/cgi/reset_password.pl').length == 3) {
+        throw FormatException(
+          'Format error, likely a wrong username or password',
+        );
+      }
+      throw FormatException(
```

Comment:
Open question: maybe we should find inspiration in @g123k's recent `SignUpStatus` (#511), that extends `Status`, in order to be able to say in all languages "your credentials are not ok".